packages/model-viewer/src/three-components/ARRenderer.ts

```diff
@@ -256,6 +292,84 @@ export class ARRenderer extends EventDispatcher<
     this.dispatchEvent({type: 'status', status: ARStatus.SESSION_STARTED});
   }
 
+  private intersectObjects(controller: any) {
+    // Do not highlight in mobile-ar
+    if (controller.userData.targetRayMode === 'screen') return;
+
+    // Do not highlight when already selected
+    if (controller.userData.selected !== undefined) return;
+
+    const line = controller.getObjectByName('line');
+    const intersections = this.getIntersections(controller);
+
+    if (intersections.length > 0) {
+      const intersection = intersections[0];
+
+      const object = intersection.object;
+      // @ts-ignore
+      object.material.emissive.setHex(0x333333);
```

Comment:
The material might already have an emissive factor - can you save this in `userData` and set back to the original instead?

Reply:
This is to highlight (brighten) the 3D model when the controller ray intersects it. Please check out the video again. ;) This highlight is removed back in the `cleanIntersected()` method by calling the `code.object.material.emissive.setHex(0x000000)`.

Reply:
Ah, I think I get it, you mean that the 3D model can have the emissive property set to a value other than 0x000000 before the controller ray intersects the 3D model? Then a question: can there be more than one 3D model in a scene at the same time?

BTW, highlighting (brightening) the 3D model on the hover is an optional improvement. I mean, we can opt out of that and simplify the implementation.

Reply:
Exactly, but this caused me to notice a potentially bigger issue. We have one 3D model, but it can be composed of an arbitrary hierarchy of objects. We don't expose those sub-meshes in our public API, so they probably should not be independently selectable. Instead we should move everything when anything is selected. And we already have a highlighting concept in WebXR: the PlacementBox - maybe best to just use that. 

I'm interested in what UX you're going for here - it might be nice to make it as similar to our existing AR UX as possible, given controllers. I'm open to other ideas, but I'd like to know how you're envisioning the experience.

Reply:
Could you please share links to an example where there is more than one 3D model in the one scene at the same time, and an example where one complex 3D model consists of multiple 3D models (parts)?

Reply:
MV only supports one 3D model, so we only have complex 3D models that consist of multiple parts. That's pretty common, e.g. https://modelviewer.dev/examples/animation/index.html

Reply:
The highlighting (brightening) the 3D model on the hover is only needed for visual feedback.

> And we already have a highlighting concept in WebXR: the PlacementBox - maybe best to just use that.

 Just tried to add `this.placementBox!.show = true` to `intersectObjects()` and `this.placementBox!.show = false` to `cleanIntersected()`. Currently, the placement box actually doesn't look like a box (cube) around the 3D model, but like a white rectangle, which is always placed horizontally at the bottom of the 3D model. It is used to select a place on the horizontal surface where to place the 3D model. When you use Meta Quest headset and select a 3D model using the ray, you are free to rotate it with your hand or controller as you want. For example, put it on its side or turn it upside down. This is something you currently can't do in AR using a smartphone and your fingers. So in its current form, the placement box is not a good fit.

Reply:
> When you use Meta Quest headset and select a 3D model using the ray, you are free to rotate it with your hand or controller as you want. For example, put it on its side or turn it upside down. This is something you currently can't do in AR using a smartphone and your fingers. 

Yes, but this is less a technical constraint than a UX decision. I think for simplicity and consistency I might prefer to limit the interactivity to Y-up as in smartphone AR. What do you think? Has anyone done UX studies on your side showing that more general interaction is valuable? The key with MV is we're not looking for "cool demo", but "easy for lay people to buy products".

Reply:
> showing that more general interaction is valuable

> easy for lay people to buy products

Instead of a thousand words. 😜

https://github.com/google/model-viewer/assets/7892779/0f45dc5d-29a2-45b8-8d0a-98175a6a1151

Reply:
As you can see in the video, a little highlighting (brightening) of the 3D model on the hover is not really needed at all. :)

Reply:
I like your style 😄. Fair point, though it makes a bit less sense with furniture. I need to think about this. Will this interaction work the same way with hands? Will we only support moving objects within arm's reach? I wish we had a UX person dedicated to thinking these use cases through, but I'm afraid it's just us for now.

Reply:
> Will this interaction work the same way with hands?

Yes, hands are also controllers in WebXR. Moreover, on Meta Quest you can switch between controllers and hands within the same XR session. You can put the controllers down and after a few seconds the controller will go to sleep and there will be an automatic switch to the hands. If you pick up the controller, it will wake up and switch back.

> Will we only support moving objects within arm's reach?

No, this approach is already essentially standard, it works with both small and large objects, both close and distant objects (the ray is long enough).

I'll try to record other videos for you to make it clearer. But it would be better if you try it yourself (use different apps and games for a few days) when you get the headset. That will clear up most of the questions. :)

> I wish we had a UX person dedicated to thinking these use cases through, but I'm afraid it's just us for now.

It's been already researched. Check out e.g. the Mixed Reality Design Language by Microsoft. ;)

Please also note that currently the user has no way to interact with the 3D model in the XR headset at all.

So my goal in this pull request is to add the simplest basic implementation. This will already improve the user experience significantly!

Then we can improve the implementation. Let's move step by step. ;)

Reply:
See https://learn.microsoft.com/en-us/windows/mixed-reality/design/point-and-commit. 😉

Reply:
Thanks a lot, that's quite helpful!

Reply:
Fixed - using the placement box as an indicator as in phone AR mode now instead of emissive materials.